network/servicenetwork/servicenetwork.go

```diff
@@ -294,3 +305,25 @@ func (n *ServiceNetwork) phaseManagerOnPulse(ctx context.Context, newPulse core.
 func isNextPulse(currentPulse, newPulse *core.Pulse) bool {
 	return newPulse.PulseNumber > currentPulse.PulseNumber && newPulse.PulseNumber >= currentPulse.NextPulseNumber
 }
+
+func (n *ServiceNetwork) verifyPulseSign(pulse core.Pulse) (bool, error) {
+	hashProvider := n.CryptographyScheme.IntegrityHasher()
+	for _, psc := range pulse.Signs {
+		payload := pulsar.PulseSenderConfirmationPayload{PulseSenderConfirmation: psc}
+		hash, err := payload.Hash(hashProvider)
+		if err != nil {
+			return false, errors.Wrap(err, "[ ServiceNetwork] verifyPulseSign: error to get a hash from pulse payload")
+		}
+		key, err := foundation.ImportPublicKey(psc.ChosenPublicKey)
```

Comment:
Do not use foundation in network code. There's same functional in `KeyProcessor` component.

Reply:
done